pkg/bytecode/code.go

```diff
@@ -42,6 +42,18 @@ const (
 	OpEqual
 	// OpNotEqual represents the inequality operator.
 	OpNotEqual
+	// OpNumLessThan represents a less than operator for numeric
+	// operands.
+	OpNumLessThan
```

Comment:
Interesting that you have `OpEqual` for all types but `OpNumLessThan`, when there is also going to be an `OpStringLessThan` I assume? To me it seems either all operators should be bound to their type or none of them, but then I guess we also have `OpAdd` and we will likely have `OpStringConcat` and `OpArrayConcat`... just musing.

Reply:
For the string and array concat, I was expecting them to be different opcodes as they are quite different operations. But comparisons are really the same thing just on different types. But that should not make any difference here - if the VM works better with specific opcodes for specific types, then it should do that. I imagine that equality/inequality was easy as `value` has an `Equals` method so there was no need for different code paths for different types.